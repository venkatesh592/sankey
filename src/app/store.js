import { configureStore } from '@reduxjs/toolkit';
import sankeyReducer from '../features/sankey/sankeySlice';

export default configureStore({
  reducer: {
    options: sankeyReducer,
  },
});
