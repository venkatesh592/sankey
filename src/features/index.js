import { DataLink } from './data-link/DataLink'
import { DataLinkList } from './data-link-list/DataLinkList'
import { DataPoint } from './data-point/DataPoint'
import { DataPointList } from './data-point-list/DataPointList'
import { EditDialog } from './dialog/EditDialog'
import { Header } from './header/Header'
import { Sankey } from './sankey/Sankey'

export {
  DataLink,
  DataLinkList,
  DataPoint,
  DataPointList,
  EditDialog,
  Header,
  Sankey
}
