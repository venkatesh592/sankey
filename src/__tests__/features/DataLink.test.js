
import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../app/store';
store.dispatch = jest.fn()
import { DataLink } from '../../features';

let container = null
beforeEach(() => {
  container = render(
    <Provider store={store}>
      <DataLink />
    </Provider>
  )
  container.findByPlaceholderText
})

test('renders With out fail', () => {
  expect(container.getByText('Link Expenditure')).toBeInTheDocument();
});

test('snapshot', () => {
  expect(container.asFragment()).toMatchSnapshot();
});

test('should input takes values - Source Placeholder', async () => {
  const element = await container.findByPlaceholderText('Source Placeholder')
  fireEvent.change(element, { target: { value: 'Salary' } })
  expect(element).toBeTruthy()
  expect(element.value).toBe('Salary')
})

test('should input takes values - Target Placeholder', async () => {
  const element = await container.findByPlaceholderText('Target Placeholder')
  fireEvent.change(element, { target: { value: 'Grocery' } })
  expect(element).toBeTruthy()
  expect(element.value).toBe('Grocery')
})

test('should input takes values - Expenditure', async () => {
  const element = await container.findByPlaceholderText('Expenditure')
  fireEvent.change(element, { target: { value: 1200 } })
  expect(element).toBeTruthy()
  expect(element.value).toBe('1200')
})

test('should create links', async () => {
  const element = await container.findByText('Create')
  fireEvent.click(element.parentElement)
  const expectedAction = {
    payload: {
      source: '',
      target: '',
      value: ''
    },
    type: 'options/setSeriesLink'
  }
  expect(store.dispatch).toHaveBeenCalledWith(expectedAction)
})

afterEach(cleanup)

afterAll(() => {
  jest.clearAllMocks()
})
