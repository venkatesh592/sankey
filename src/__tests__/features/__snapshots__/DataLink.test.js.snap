// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="dataLink"
  >
    <span>
      Link Expenditure
    </span>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline input"
    >
      <input
        class="MuiInputBase-input MuiInput-input"
        placeholder="Source Placeholder"
        type="text"
        value=""
      />
    </div>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline input"
    >
      <input
        class="MuiInputBase-input MuiInput-input"
        placeholder="Target Placeholder"
        type="text"
        value=""
      />
    </div>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline input"
    >
      <input
        class="MuiInputBase-input MuiInput-input"
        placeholder="Expenditure"
        type="text"
        value=""
      />
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Create
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
